qiskit/compiler/assemble.py

```diff
@@ -289,20 +292,33 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
     if rep_time:
         if dynamic_reprate_enabled:
             warnings.warn("Dynamic rep rates are supported on this backend. 'rep_delay' will be "
-                          "used instead, if specified.", RuntimeWarning)
+                          "used instead of 'rep_time'.", RuntimeWarning)
         if isinstance(rep_time, list):
             rep_time = rep_time[0]
         rep_time = int(rep_time * 1e6)  # convert sec to μs
 
-    rep_delay = rep_delay or getattr(backend_config, 'rep_delays', None)
-    if rep_delay:
-        if not dynamic_reprate_enabled:
-            warnings.warn("Dynamic rep rates not supported on this backend. 'rep_time' will be "
-                          "used instead.", RuntimeWarning)
-
-        if isinstance(rep_delay, list):
-            rep_delay = rep_delay[0]
-        rep_delay = rep_delay * 1e6  # convert sec to μs
+    if dynamic_reprate_enabled:
+        rep_delay = rep_delay or getattr(backend_config, "default_rep_delay", None)
+        if rep_delay is not None:
+            rep_delay_range = getattr(backend_config, "rep_delay_range", [])
```

Comment:
This should not perform this check if the rep_delay_range is not set. Right now its going to raise a very confusing error about the length being wrong.

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/4830/commits/9a76631d237dc68bc27bd7be4a6654dc2483cef0 and https://github.com/Qiskit/qiskit-terra/pull/4830/commits/bb96fc21cc75386b7396c65af1aa302d48d420dd.